GitOut/Features/Git/LocalGitRepository.cs

```diff
@@ -30,6 +30,16 @@ IProcessFactory<IGitProcess> processFactory
 
         public GitStatusResult? CachedStatus { get; private set; }
 
+        public async Task<bool> IsInsideGitFolder()
+        {
+            IGitProcess proc = CreateProcess(ProcessOptions.Builder().AppendRange("rev-parse", "--is-inside-work-tree").Build());
+            await foreach (string line in proc.ReadLinesAsync())
```

Comment:
Do we need to do this async and with a foreach? Would a `ReadAll..` suffice?

Reply:
This is currently how the GitProcess class operates, by returning an IAsyncEnumerable.
I could change this to instead be
```cs
IGitProcess proc = CreateProcess(ProcessOptions.Builder().AppendRange("rev-parse", "--is-inside-work-tree").Build());
var result = await proc.ExecuteAsync();
return result.OutputLines.Count == 1 && result.OutputLines.First() == "true";
```
But I don't know if that is any better in terms of readability :innocent:
Performance-wise they should do equally well from what I can guess

Reply:
I'm rather suggesting to not do it asynchronously. Do we expect this to ever taking longer than a few milliseconds?
```csharp
IGitProcess proc = CreateProcess(ProcessOptions.Builder().AppendRange("rev-parse", "--is-inside-work-tree").Build());
var result = proc.ExecuteSingle();
return result.OutputLine == "true";
```

Reply:
We must do it more or less async, GitProcess spawns an executable and we need to wait for its response (in the form of event callbacks which is feed back async atm) so I don't think we can do it sync unless we do some extra work with wait handles of some kind